code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/AepNotificationUtil.kt

```diff
@@ -0,0 +1,178 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.ui.notification.builders.AutoCarouselNotificationBuilder
+import com.adobe.marketing.mobile.services.ui.notification.builders.BasicNotificationBuilder
+import com.adobe.marketing.mobile.services.ui.notification.builders.LegacyNotificationBuilder
+import com.adobe.marketing.mobile.services.ui.notification.builders.ManualCarouselNotificationBuilder
+import com.adobe.marketing.mobile.services.ui.notification.models.AepPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.AutoCarouselPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.BasicPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.CarouselPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.ManualCarouselPushTemplate
+
+/**
+ * Public facing object to construct a [NotificationCompat.Builder] object for the specified [PushTemplateType].
+ * The [constructNotificationBuilder] methods will build the appropriate notification based on the provided
+ * [AepPushTemplate] or [Intent].
+ */
+object AepNotificationUtil {
+    private const val SELF_TAG = "AepNotificationUtil"
+
+    @JvmStatic
+    val notificationCompatPriorityMap: Map<Int, String> = mapOf(
```

Comment:
this is used on the caller side like:
```
AepNotificationUtil.getNotificationCompatVisibilityMap().get(notification.getVisibility()));
```
in a situation where the caller needs to convert priority or visibility. these maps can be stored on the caller extension side. removing them.

Reply:
The constant values to create this mapping on the caller side can't be accessed since `AEPPushTemplate` is an internal seal class and `NotificationPriority` is an internal object.  The constant values are also used internally in `AEPPushTemplate` class. Can we keep this here?

Reply:
No longer a problem after moving `NotificationPriority` to `PushTemplateConstants` class

Reply:
@prudrabhat can we add this back?

Reply:
Added back as public enums